### ISSUE TITLE
FaceGrid.map and FacetGrid.map_dataframe can alter legend_out

### ISSUE THREAD

**Issue by lewisamarshall** [mwaskom/seaborn#1580]
In 0.9.0, using the FacetGrid.map or FacetGrid.map_dataframe functions breaks the alignment on legend_out=True. This is shown in the plot below, in which the "grouped_barplot.py" example has g.map() called with a dummy function before rendering.

![original_grouped_barplot](https://user-images.githubusercontent.com/3657774/46553342-20732e80-c892-11e8-8724-ff29a758dc79.png)

This is came from the function _finalize_grid, which was calling tight_layout as a final step. Removing that line prevents resizing into the legend.

![new_grouped_barplot](https://user-images.githubusercontent.com/3657774/46553364-28cb6980-c892-11e8-99da-2b206ceb3bfc.png)

System:
OS: OSX
Python: Python 3.6.6 :: Anaconda custom (64-bit)
Matplotlib: 2.2.2
Seaborn: 0.9.0

Code to reproduce:

```
import seaborn as sns

sns.set(style="whitegrid")

titanic = sns.load_dataset("titanic")

g = sns.catplot(x="class", y="survived", hue="sex", data=titanic,
height=6, kind="bar", palette="muted")
g.despine(left=True)
g.set_ylabels("survival probability")

def f(*args, **kwargs):
    pass

g.map(f)
```

**Comment by fakufaku** [comment]
I'm having the same problem on Mac OS and Linux too.

**Comment by herrlich10** [comment]
Currently, one way to work around this issue is `add_legend()` after `g.map()`.

    g = sns.catplot(..., legend=False)
    g.map(...)
    g.add_legend()

**Comment by mwaskom** [comment]
Closed with #2073
